### DISCUSSION TITLE
Socket exception: Bad file descriptor (9) with scrapli via paramiko transport

### DISCUSSION THREAD

**Discussion by igor-a-silkin** [carlmontanari/scrapli#236] [Q&A] [answered]
Hi,

I'm newbie Python. I'm using `scrapli` via the paramiko transport to connect to old Cisco devices and am getting the following error:
`Socket exception: Bad file descriptor (9)`

My fragment code is next:
```
...
r1 = {
    "host": "192.168.20.20",
    "auth_username": "cisco",
    "auth_password": "cisco",
    "auth_secondary": "cisco",
    "auth_strict_key": False,
    "platform": "cisco_iosxe",
    "transport": "paramiko",
}

def send_show(device, show_commands):
    if type(show_commands) == str:
        show_commands = [show_commands]
    cmd_dict = {}
    
    try:
        with Scrapli(**device) as ssh:
            for cmd in show_commands:
                reply = ssh.send_command(cmd)
                cmd_dict[cmd] = reply.result
    except ScrapliException as error:
        print(error, device["host"])
    return cmd_dict
...
```
The error occurs after the `send_show` function. The code works, but this error confuses.

Tell me, please, what am I doing wrong?

**Comment by carlmontanari** [comment, marked as answer]
Hey @igor-a-silkin -- you are doing nothing wrong! I've seen this pop up and it seems to be cosmetic so I never bothered looking further into it.... if you spend any time and find a root cause I'm very open to a PR but I probably will not dig into it any time soon.

**Reply by igor-a-silkin** [reply]
Hey [carlmontanari](@carlmontanari), thanks for your reply.

**Comment by KlimchevKonstantin** [comment]
I also encountered such a situation and I had the opinion that the exception occurs when trying to close an already closed connection


For example:
```
conn = someDriver()
conn.open()
...
conn.close()
```
sometimes occurs  

and at
```
conn = someDriver()
conn.open()
...
```
never encountered (without close())

I can’t say for sure, because I’m currently using ssh2 transport
